python/orion_finance_sdk_py/contracts.py

```
@@ -804,15 +972,168 @@ def share_price(self) -> int:
+                (ints ``>= 1_000_000_000`` are treated as timestamps).
+            end: End bound (same types as ``start``). Defaults to latest block.
+            interval: Sampling interval. Only ``"1d"`` (daily) is supported for now.
+
+        Returns:
+            List of ``{"timestamp", "block", "share_price"}`` dicts (unix timestamp,
+            block number, share price in underlying units).
+        """
+        if interval != "1d":
+            raise ValueError(
+                f"Unsupported interval {interval!r}. Only '1d' is supported."
+            )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)